### PR TITLE
feat(node/net): unix domain socket support

### LINKED ISSUE #1936: Need explanation, net node module return Not implemented error
I'm trying to understand the issue about `Not implemented` error on net module, so I could try to fix an issue with [web3](https://deno.land/x/web3@v0.10.0) package

Here the code and the error messages
```js
import net from "https://deno.land/std@0.125.0/node/net.ts";
import Web3 from "https://deno.land/x/web3/mod.ts";

const provider = new Web3.providers.IpcProvider(
      "/root/geth_data/geth.ipc",
      net
    );
const web3 = new Web3(provider);
```

```
error: Uncaught (in promise) Error: Not implemented
  throw new Error(message);
        ^
    at notImplemented (https://deno.land/std@0.125.0/node/_utils.ts:23:9)
    at Pipe.connect (https://deno.land/std@0.125.0/node/internal_binding/pipe_wrap.ts:93:5)
    at _internalConnect (https://deno.land/std@0.125.0/node/net.ts:470:36)
    at defaultTriggerAsyncIdScope (https://deno.land/std@0.125.0/node/_async_hooks.ts:57:18)
    at Socket.connect (https://deno.land/std@0.125.0/node/net.ts:926:7)
    at Object.connect (https://deno.land/std@0.125.0/node/net.ts:1580:17)
    at new IpcProvider (https://deno.land/x/web3@v0.9.2/packages/web3-providers-ipc/src/index.js:35:25)
```

**Expected behavior**
No error, web3 will connect normally to the ipc

**Environment**

- OS: Ubuntu 20.04
- deno version: 1.18.2
- std version: 0.125.0

### PR DESCRIPTION
**Pull request by cmorten**
Works towards:

- #1936 
- #1635 

Notes:

- Doesn't add windows support as blocked by https://github.com/denoland/deno/issues/10244.
- Can't complete the full interface as there is outstanding work needed into Deno core.
- Improved test coverage for net again after lost from streams rework. Number of tests not passing so more work needed here.
- Couldn't get some `path` related tests passing - unsure if an issue with existing net/streams code or missing something in the new work here... but perhaps this is a reasonable start?
- Files changed looks daunting, but mostly a lot of Node compat tests. Unfortunately formatting updates find the main changes to some files, but hopefully straight-forward reading.

### CODE REVIEW

File: node/internal_binding/cares_wrap.ts [hunk 1]
```diff
@@ -92,12 +92,14 @@ export function getaddrinfo(
 
     const error = addresses.length ? null : codeMap.get("EAI_NODATA")!;
 
+    // TODO(cmorten): needs work
+    // REF: https://github.com/nodejs/node/blob/master/src/cares_wrap.cc#L1444
```
[the remark this answers is not in the document]
Comment (cmorten) [reply node/internal_binding/cares_wrap.ts:96]: Covered in coversation here https://github.com/denoland/deno/issues/10762, though looks like never got addressed...

**Review by kt3k — APPROVED** [review]
LGTM!

Great first pass for ipc conn support in node compat! Also thank you for enabling lot of compat test cases!

**Comment by kt3k** [comment]
Confirmed a basic use case working

In node (compat)
```js
const net = require("net");
net.createServer((c) => { 
  c.on("data", (data) => { 
    console.log("on data", data.toString());
    c.end("HTTP/1.1 200 OK\nContent-type: text/html\n\nhello\n");
  });
}).listen("/path/to/my.socket", () => { 
  console.log("server linstening at localhost:3001");
});
```

Requesting it from deno
```js
const hostname = "example.com"
const conn = await Deno.connect({ path: "/path/to/my.socket", transport: "unix" }); 
const blob = new Blob([`GET / HTTP/1.1
Host: ${hostname}

`]);
const writer = conn.writable.getWriter();
writer.write(new Uint8Array(await blob.arrayBuffer()));
conn.readable.pipeTo(Deno.stdout.writable);
```

File: node/internal_binding/pipe_wrap.ts [hunk 2]
```diff
@@ -132,12 +133,12 @@ export class Pipe extends ConnectionWrap {
...
     }
 
-    const connectOptions: Deno.UnixConnectOptions = {
+    const connectOptions: DenoUnstable.UnixConnectOptions = {
       path: address,
       transport: "unix",
     };
 
-    Deno.connect(connectOptions).then(
+    DenoUnstable.connect(connectOptions).then(
```
Comment (kt3k) [inline node/internal_binding/pipe_wrap.ts:141]: This is helpful. Thanks!

**Review by kt3k — APPROVED** [review]
LGTM again

**Pull request merged**